tests/Doctrine/Tests/ORM/Functional/PaginationTest.php

```diff
@@ -667,6 +667,27 @@ public function testPaginationWithSubSelectOrderByExpression($useOutputWalker, $
         self::assertCount(9, $paginator->getIterator());
     }
 
+    public function testDifferentResultLengthsDoNotRequireExtraQueryCacheEntries(): void
+    {
+        $dql   = 'SELECT u FROM Doctrine\Tests\Models\CMS\CmsUser u WHERE u.id >= :id';
+        $query = $this->_em->createQuery($dql);
+        $query->setMaxResults(10);
+
+        $query->setParameter('id', 1);
+        $paginator = new Paginator($query);
+        count($paginator->getIterator()); // exercise the Paginator
```

Comment:
`getIterator()` returns an `ArrayIterator`, but only declares a `Traversable`, so I'd use `iterator_count()` here and below, to be on the safe side.

Reply:
Ah, we don't need that at all, I'll fix it.